controller_manager/src/ros2_control_node.cpp

```diff
@@ -41,17 +41,19 @@ int main(int argc, char ** argv)
     RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
 
     rclcpp::Time begin = cm->now();
+    rclcpp::Time begin_last = begin;
+    rclcpp::Time end = begin;
 
     // Use nanoseconds to avoid chrono's rounding
     std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000 / cm->get_update_rate()));
     while (rclcpp::ok())
     {
-      rclcpp::Time begin_last = begin;
       begin = cm->now();
       cm->read();
```

Comment:
I think we should rather use something like this:

```suggestion
      cm->read();
      begin = cm->now();
```
In this case, only update method is not counted in the period. This is not 100% exact when having multiple controller but closer to the real value.